src/DeviceInput/Implementations/webDeviceInputSystem.ts

```diff
@@ -420,17 +429,30 @@ export class WebDeviceInputSystem implements IDeviceInputSystem {
                 this._addPointerDevice(deviceType, deviceSlot, evt.clientX, evt.clientY);
             }
 
-            if (deviceType === DeviceType.Mouse && this._mouseId === -1) {
-                this._mouseId = evt.pointerId;
-            }
-
             const pointer = this._inputs[deviceType][deviceSlot];
             if (pointer) {
                 const previousHorizontal = pointer[PointerInput.Horizontal];
                 const previousVertical = pointer[PointerInput.Vertical];
                 const previousButton = pointer[evt.button + 2];
 
-                this._elementToAttachTo.setPointerCapture(evt.pointerId);
+                if (deviceType === DeviceType.Mouse && this._mouseId === -1) { // Mouse; Among supported browsers, value is either 1 or 0 for mouse
+                    if (evt.pointerId === undefined) { // If there is no pointerId (eg. manually dispatched MouseEvent)

```

Comment:
why not fixing it at the SimulatePointerDOwn level? This would avoid having to generate id that could collide with real ones

Reply:
I've thought about this but I decided to go this route because both the InputManager's and DeviceSourceManager's input routes through the DeviceInputSystem.  If I were to fix it at the simulatePointerDown level, it would still be broken if anyone were to use MouseEvents with the DeviceSourceManager.

With the interpreting of the pointerId, we're only doing that for the mouse and in my various tests, the id for mouse is 1 with the exception of Firefox.  According to the spec, the default value for pointerId is 0 so if the event doesn't have a properly defined pointerId (either _undefined_ or the default value for a non-FF browser), we'd have to adjust it or an exception will be thrown.  If the PointerEvent is properly created, this interpretation should never happen.

Reply:
ok I'm just not at ease with sniffing FF. BUt not sure how to avoid it